data/transactions/logic/eval.go

```diff
@@ -2664,7 +2718,7 @@ func opEcdsaVerify(cx *EvalContext) {
 		return
 	}
 
-	if fs.field != Secp256k1 {
+	if fs.field != Secp256k1 && fs.field != Secp256r1 {
```

Comment:
I think we should have a version assertion here.

Reply:
I can't see a reason for this entire `if` clause (here, or in the other opEcdsa functions).  Doesn't the previous `!ok` check handle it?

Reply:
We should ensure that we have a unit test that tests behavior when the field is wrong.  This is somewhat hard to write: You have to compile the TEAL, then edit the byte to be wrong, since the assembler isn't going to let you create the bytecode with an invalid field number.  We have some tests like that though.

If it is caught by normal evaluation (it is unless I'm reading wrong), I'd really not like to see us checking the same thing that was checked two lines earlier.